### PR TITLE
New Learn Use Case: Creating Customer Analytics ML Pipeline with Snowpark, Weaviate, and Streamlit

### PR DESCRIPTION
**Pull request by astronautyates**
Can you please provide guidance on how to add images to repo? Looks like old /img directory got removed, and old links are broken so would you mind letting me know the new format if this is intentional and I'll add screenshots?

### CODE REVIEW

**Comment by TJaniF** [comment]
> Can you please provide guidance on how to add images to repo? Looks like old /img directory got removed, and old links are broken so would you mind letting me know the new format if this is intentional and I'll add screenshots?

Hey :) 
I've been adding images for use cases under `static/img/examples` and tried to always have the id of the learn document as the first part of the name so the images are easier to find, but yes I keep thinking one day I should clean that up, some older use cases have a couple of images in guides or learn 😅

**Comment by astronautyates** [comment]
Thanks @TJaniF ! I added my images in a new dedicated folder, not sure how to stop the learn folder from being added and breaking the reference to the images though, any suggestions?

**Comment by TJaniF** [comment]
> Thanks @TJaniF ! I added my images in a new dedicated folder, not sure how to stop the learn folder from being added and breaking the reference to the images though, any suggestions?

Hmmm weird. Maybe if you remove the "static" in the link like this:

```
/img/use_case_images/snowpark_weaviate_use_case/use-case-snowpark_weaviate_ca_dag_2.png
```

**Comment by astronautyates** [comment]
Fixed it, needed a slash at the start of the file path...

File: learn/use-case_snowpark_customer_analytics.md [hunk 1]
```diff
@@ -0,0 +1,906 @@
+---
+title: "Production-ready ML with Airflow, Snowpark, and Weaviate"
+description: "Use Airflow, Snowpark and Weaviate to use Customer sentiment to predict LTV"
+id: use-case-ml-airflow-snowpark-weaviate
+sidebar_label: "ML with Snowpark and Airflow"
+sidebar_custom_props: { icon: 'img/integrations/snowpark.png' }
+---
+
+![A screenshot of the streamlit application created in this use case that shows the customer analytics dashboard](/static/img/use_case_images/snowpark_weaviate_use_case/use-case-snowpark_weaviate_streamlitdashboardss1.png)
```
Comment (kentdanas) [inline learn/use-case_snowpark_customer_analytics.md:9]:
```suggestion
![A screenshot of the streamlit application created in this use case that shows the customer analytics dashboard](/img/use_case_images/snowpark_weaviate_use_case/use-case-snowpark_weaviate_streamlitdashboardss1.png)
```
I think this should fix the broken checks.

File: learn/use-case_snowpark_customer_analytics.md [hunk 2]
```diff
@@ -0,0 +1,906 @@
...
+
+This guide demonstrates how to use Apache Airflow to orchestrate a machine learning pipeline leveraging the Snowpark provider and Snowpark ML for feature engineering and model tracking. While Snowpark ML has its own support for models similar to scikit-learn this code demonstrates a "bring-your-own" model approach showing the use of open-source scikit-learn along with Snowpark ML model registry and model serving in an Airflow task rather than Snowpark user-defined function (UDF).  
+
+![A screenshot of the Customer Analytics DAG](/static/img/use_case_images/snowpark_weaviate_use_case/use-case-snowpark_weaviate_ca_dag_2.png)
```
Comment (kentdanas) [inline learn/use-case_snowpark_customer_analytics.md:15]:
```suggestion
![A screenshot of the Customer Analytics DAG](/img/use_case_images/snowpark_weaviate_use_case/use-case-snowpark_weaviate_ca_dag_2.png)
```

File: learn/use-case_snowpark_customer_analytics.md [hunk 3]
```diff
@@ -0,0 +1,906 @@
...
+
+## Set-up the project
+
+Clone the example project from the [Astronomer GitHub](https://github.com/astronomer/airflow-snowparkml-demo/tree/main).
+
+```bash
+git clone https://github.com/astronomer/airflow-snowparkml-demo
+cd airflow-snowparkml-demo 
+```
+![A screenshot of the environment file structure](/static/img/use_case_images/snowpark_weaviate_use_case/use-case-snowpark_weaviate_filestructure_3.png)
```
Comment (kentdanas) [inline learn/use-case_snowpark_customer_analytics.md:51]:
```suggestion
![A screenshot of the environment file structure](/img/use_case_images/snowpark_weaviate_use_case/use-case-snowpark_weaviate_filestructure_3.png)
```

File: learn/use-case_snowpark_customer_analytics.md [hunk 4]
```diff
@@ -0,0 +1,906 @@
...
+NOTE: Database and Schema names should be CAPITALIZED due to a bug in Snowpark ML.
+
+## Run the project
+
+To run the example project, first make sure Docker Desktop is running. Then, open your project directory and run:
+
+```sh
+astro dev start
+```
+![A screenshot of the Airflow UI with active Customer Analytics DAG](/static/img/use_case_images/snowpark_weaviate_use_case/use-case-snowpark_weaviate_activeplaybutton_4.png)
```
Comment (kentdanas) [inline learn/use-case_snowpark_customer_analytics.md:75]:
```suggestion
![A screenshot of the Airflow UI with active Customer Analytics DAG](/img/use_case_images/snowpark_weaviate_use_case/use-case-snowpark_weaviate_activeplaybutton_4.png)
```

File: learn/use-case_snowpark_customer_analytics.md [hunk 5]
```diff
@@ -0,0 +1,906 @@
...
+
+Then, run the following command to start Streamlit
+```bash
+cd include/streamlit/src
+python -m streamlit run ./streamlit_app.py
+```
+
+You can then open the [streamlit application](http://localhost:8501) in a browser. All of the customer analytics produced by your DAG will be visualized here, and if you've added an Open-AI key to your .env file, you'll be able to leverage the chatbot functionality to search for customer reviews by keyword as depicted below: 
+
+![A screenshot of the Airflow UI with active Customer Analytics DAG](/static/img/use_case_images/snowpark_weaviate_use_case/use-case-snowpark_weaviate_chatinteraction_streamlit_5.png.png)
```
Comment (kentdanas) [inline learn/use-case_snowpark_customer_analytics.md:99]:
```suggestion
![A screenshot of the Airflow UI with active Customer Analytics DAG](/img/use_case_images/snowpark_weaviate_use_case/use-case-snowpark_weaviate_chatinteraction_streamlit_5.png.png)
```

**Pull request closed without merging**